sim_pipeline/halos/halos_lens.py

```diff
@@ -0,0 +1,196 @@
+
+    Attributes
+    ----------
+    halos_list : table
+        List of halos for which the lensing properties are computed.
+    n_halos : int
+        The number of halos in `halos_list`.
+    sky_area : float
+        The total sky area in deg^2 over which the halos are distributed.
+    redshift_list : array_like
+        The redshifts of the halos.
+    mass_list : array_like
+        The masses of the halos.
+    cosmo : astropy.Cosmology instance
+        The cosmology assumed in the lensing computations.
+
+    Methods
+    -------
+    set_lens_model() :
+        Returns a LensModel instance for the NFW profile for all halos.
+    random_position() :
+        Returns random x and y coordinates in the sky, uniformly distributed within a circular sky area.
+    get_nfw_kwargs() :
+        Returns the scale radius, observed bending angle at the scale radius, and positions of the halos
+        in the lens plane.
+    get_halos_lens_kwargs() :
+        Returns the list of keyword arguments for each halo to be used in the lens model.
+    get_convergence_shear() :
+        Returns the convergence and shear at the origin due to all the halos.
+    """
+    # TODO: ADD test functions
+    def __init__(self, halos_list, cosmo=None, sky_area=4 * np.pi):
+        self.halos_list = halos_list
+        self.n_halos = len(self.halos_list)
+        self.sky_area = sky_area
+        self.redshift_list = halos_list['z']
+        self.mass_list = halos_list['mass']
+        if cosmo is None:
+            warnings.warn("No cosmology provided, instead uses astropy.cosmology import default_cosmology")
+            from astropy.cosmology import default_cosmology
+            self.cosmo = default_cosmology.get()
+
+    def set_lens_model(self):
+        """
+        Initializes and returns a LensModel object with a Navarro-Frenk-White (NFW) profile for every halo.
+
+        Returns
+        -------
+        lens_model : lenstronomy.LensModel instance
+            The LensModel object with a NFW profile for every halo.
+        """
+        lens_model = LensModel(lens_model_list=['NFW'] * self.n_halos,
+                               lens_redshift_list=self.redshift_list,
+                               cosmo=self.cosmo,
+                               observed_convention_index=[],
+                               multi_plane=True,
+                               z_source=5
+                               )
+        # TODO: Set z_source as an input parameter or other way
+        return lens_model
+
+    def random_position(self):
+        """
+        Generates and returns random positions in the sky using a uniform distribution.
+
+        Returns
+        -------
+        px, py : float
+            The generated random x and y coordinates inside the skyarea in arcsec.
+        """
+        phi = 2 * np.pi * np.random.random()
+        upper_bound = np.sqrt(self.sky_area / np.pi)
+        random_radius = 3600 * np.random.uniform(0, upper_bound)
```

Comment:
does this render a uniform distribution within a disk? Uniform in radius is not uniform in area

Reply:
uniform distribution for radius and phi for angle, combined for uniform in area

Reply:
> does this render a uniform distribution within a disk? Uniform in radius is not uniform in area

uniform distribution for radius and phi for angle, combined for uniform in area

Reply:
I am confused, uniform in radius will lead to non-uniform in area. See e.g. here how you can render uniform in area within a radius: https://github.com/LSST-strong-lensing/sim-pipeline/blob/1ef82e2094e5a9e4a6f42fcdf66acc0aae84b2d7/sim_pipeline/gg_lens.py#L78

Reply:
this might also be part of the problem as you sample too many halos too close to the center